unitycatalog/functions/client.py

```diff
@@ -0,0 +1,136 @@
+import json
+import logging
+import threading
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Dict, Literal, Optional
+
+_logger = logging.getLogger(__name__)
+
+_uc_function_client = None
+_client_lock = threading.Lock()
+
+
+@dataclass
+class FunctionExecutionResult:
+    """
+    Result of executing a function.
+    Value is always string, even if the function returns a scalar.
+    """
+
+    error: Optional[str] = None
+    format: Optional[Literal["SCALAR", "CSV"]] = None
+    value: Optional[str] = None
+    truncated: Optional[bool] = None
+
+    def to_json(self) -> str:
+        data = {k: v for (k, v) in self.__dict__.items() if v is not None}
+        return json.dumps(data)
+
+
+class BaseFunctionClient(ABC):
+    """
+    Base class for uc function calling client
+    """
+
+    def __init__(self, client: Any = None, **kwargs: Any) -> None:  # noqa: ARG002
+        self._lock = threading.Lock()
+
+    @abstractmethod
+    def create_function(self, function_info: Any) -> Any:
+        """Create a function"""
+        raise NotImplementedError
+
+    @abstractmethod
+    def retrieve_function(self, function_name: str, **kwargs: Any) -> Any:
```

Comment:
```suggestion
    def get_function(self, function_name: str, **kwargs: Any) -> Any:
```

for easier and faster typing.